server/src/main/java/org/apache/cassandra/sidecar/utils/SidecarClientProvider.java

```diff
@@ -0,0 +1,166 @@
+    {
+        LOGGER.info("Closing Sidecar Client...");
+        try
+        {
+            client.close();
+            Thread.sleep(100);
+            completion.complete();
+        }
+        catch (Throwable throwable)
+        {
+            completion.fail(throwable);
+        }
+    }
+
+    private SidecarClient initializeSidecarClient()
+    {
+        WebClientOptions webClientOptions = webClientOptions();
+        HttpClient httpClient = vertx.createHttpClient(webClientOptions);
+        WebClient webClient = WebClient.wrap(httpClient, webClientOptions);
+
+        HttpClientConfig httpClientConfig = new HttpClientConfig.Builder<>()
+                                            .ssl(webClientOptions().isSsl())
+                                            .timeoutMillis(clientConfig.requestTimeout().toMillis())
+                                            .idleTimeoutMillis(clientConfig.requestIdleTimeout().toIntMillis())
+                                            .userAgent("cassandra-sidecar/" + sidecarVersionProvider.sidecarVersion())
+                                            .build();
+
+        VertxHttpClient vertxHttpClient = new VertxHttpClient(vertx, webClient, httpClientConfig);
+        RetryPolicy defaultRetryPolicy = new ExponentialBackoffRetryPolicy(clientConfig.maxRetries(),
+                                                                           clientConfig.retryDelay().toMillis(),
+                                                                           clientConfig.retryDelay().toMillis());
+        SidecarClientVertxRequestExecutor requestExecutor = new SidecarClientVertxRequestExecutor(vertxHttpClient);
+        SidecarInstance instance = new SidecarInstanceImpl(webClientOptions.getDefaultHost(), webClientOptions.getDefaultPort());
+        ArrayList<SidecarInstance> instances = new ArrayList<>();
+        instances.add(instance);
+        SimpleSidecarInstancesProvider instancesProvider = new SimpleSidecarInstancesProvider(instances);
+
+        SidecarClientConfig config = SidecarClientConfigImpl.builder()
+                                                            .retryDelayMillis(clientConfig.retryDelay().toMillis())
+                                                            .maxRetryDelayMillis(clientConfig.maxRetryDelay().toMillis())
+                                                            .maxRetries(clientConfig.maxRetries())
+                                                            .build();
+
+        return new SidecarClient(instancesProvider,
+                                 requestExecutor,
+                                 config,
+                                 defaultRetryPolicy);
+    }
+
+    private WebClientOptions webClientOptions()
+    {
+        WebClientOptions options = new WebClientOptions();
+        options.getPoolOptions()
+               .setCleanerPeriod(clientConfig.connectionPoolCleanerPeriod().toIntMillis())
```

Comment:
We should add these client settings to `HttpClientConfig`, that way we dont have to create `WebClientOptions separately`, also `VertxHttpClient` takes care of setting ssl config, we can use it